### PR TITLE
fix: rendering the circular items

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:

Rendering the circular items.

image

**Related issue(s)**
Fixes https://github.com/asyncapi/asyncapi-react/issues/351

### CODE REVIEW

File: library/src/components/Schema.tsx [hunk 1]
```diff
@@ -38,17 +38,31 @@ export const Schema: React.FunctionComponent<Props> = ({
...
+
+  let uid = schema.uid();
+
+  // checking uid for circular items
+  // after fixing https://github.com/asyncapi/parser-js/issues/293 statement should be removed
+  if (
+    isCircular &&
+    !uid &&
+    schema.items() &&
+    typeof (schema.items() as SchemaType).uid === 'function'
```
Comment (derberg) [inline library/src/components/Schema.tsx:61]: how come uid could be a function?

Comment (magicmatatjahu) [reply library/src/components/Schema.tsx:61]: Items can be also array of schemas, like tuple type, so for that I used `typeof (schema.items() as SchemaType).uid === 'function'` to check that I operate on schema, not on array of schemas.

Comment (magicmatatjahu) [reply library/src/components/Schema.tsx:61]: Also `uid` is used here to check what exactly schema is circular

Comment (derberg) [reply library/src/components/Schema.tsx:61]: > Items can be also array of schemas, like tuple type, so for that I used typeof (schema.items() as SchemaType).uid === 'function' to check that I operate on schema, not on array of schemas.

this is not that straight forward, can you add a comment to code please 🙏🏼 

> Also uid is used here to check what exactly schema is circular

I don't know what you mean. Tbh I don't remember how it is handled on the parser side. You mean that circular schema has no `uid`?

Comment (magicmatatjahu) [reply library/src/components/Schema.tsx:61]: > You mean that circular schema has no uid?

Circular schemas have uid, but problem with `items` is that `uid` is added to schema in `items` field but `x-parser-circular` info is added to parent schema, which has `items` field 😅 So I must do this "tricky" think here - I have info that schema has circular ref in items, but I must read uid from `items()` function

Comment (derberg) [reply library/src/components/Schema.tsx:61]: 🤯 
this one definitely deserves a code comment, way to deep knowledge about the parser needed to get the code 😄

Comment (magicmatatjahu) [reply library/src/components/Schema.tsx:65]: Added :)

**Review by derberg — APPROVED** [review]

**Pull request merged**